File: Documentation/5.4/Samples/python/ClientApi/Session/HowTo/GetCountersFor.py
from typing import List

from examples_base import ExampleBase


class GetCountersFor(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_get_counters_for(self):
        with self.embedded_server.get_document_store("GetCountersFor") as store:
            with store.open_session() as session:
                # region example
                # Load a document
                employee = session.load("employees/1-A")

                # Get counter names from the loaded entity
                counter_names = session.advanced.get_counters_for(employee)
                # endregion

    class Foo:
        # region syntax
        def get_counters_for(self, entity: object) -> List[str]: ...

        # endregion

File: Documentation/5.4/Samples/php/ClientApi/session/HowTo/GetCountersFor.php
<?php

namespace RavenDB\Samples\ClientApi\Session\HowTo;

use Employee;
use RavenDB\Documents\DocumentStore;
use RavenDB\Type\StringList;

interface FooInterface
{
    # region syntax
    public function getCountersFor(mixed $instance): ?StringList;
    # endregion
}

class GetCountersFor
{
    public function example(): void
    {
        $store = new DocumentStore();
        try {
            $session = $store->openSession();
            try {
                # region example
                // Load a document
                $employee = $session->load(Employee::class, "employees/1-A");

                // Get counter names from the loaded entity
                $counterNames = $session->advanced()->getCountersFor($employee);
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
